lib/ForkProcessSandbox.cpp

```diff
@@ -0,0 +1,48 @@
+
+#include "ForkProcessSandbox.h"
+
+#include "TestResult.h"
+
+#include <sys/mman.h>
+#include <unistd.h>
+
+Mutang::ExecutionResult Mutang::ForkProcessSandbox::run(std::function<void (ExecutionResult *)> Func) {
+
+  void *SharedMemory = mmap(NULL,
+                            sizeof(ExecutionResult),
+                            PROT_READ | PROT_WRITE,
+                            MAP_SHARED | MAP_ANONYMOUS,
+                            -1,
+                            0);
+
+  ExecutionResult *SharedResult = new (SharedMemory) ExecutionResult();
+
+  int ChildPid = fork();
+  if (ChildPid == -1) {
+    exit(1);
+  }
+
+  else if (ChildPid == 0) {
+    Func(SharedResult);
+    exit(0);
+  }
+
+  else {
+    /// TODO: Handle at least 3 cases:
+    /// 1) Child process crashes
+    /// 2) Child process exits 1
+    /// 3) Child process hangs
```

Comment:
> I wish we could rely on it, I'm afraid we will have to implement the timeouts just for our own safety.
> But we will see, I'm not 100% sure.

Maybe you want both, but consider a timeout hit from the mutation testing engine a bug. Assumption: If the test framework could not enforce a timeout, when executed from mutation testing, something is really wrong with the integration.

This is at least the path mutant will do in future, test timeouts in ruby are possible but not 100% solid.